libmscore/segment.cpp

```diff
@@ -549,7 +549,11 @@ void Segment::add(Element* el)
                   _elist[track] = el;
                   _empty = false;
                   break;
-
+            case Element::Type::ANNOTATION:
+                  checkElement(el, track);
+                  _elist[track] = el;
```

Comment:
This isn't the right way to add an annotation.  Look further up - see how staff texts (and pretty much all the other texts) are added, by adding to the "_annotations" list for the segment?  You should probably just add this case to that list of other cases.  Then the annotation will be in the list of segment annotations (I know, two different uses of the word annotation, confusing, but probably not worth changing at this point).

Anyhow, once this is done, then the change I suggested to explicitly call layout when doing other segment annotations should work.